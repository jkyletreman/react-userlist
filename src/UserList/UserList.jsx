import React from "react";
import UserData from "../data/context";

export const UserList = () => {
  return (
    <UserData.Consumer>
      {context => {
        return context.state.userData.map(user => {
          return (
            <li key={user.id} onClick={context.selectUser}>
              {user.first} {user.last}
            </li>
          );
        });
      }}
    </UserData.Consumer>
  );
};

export default UserList;
