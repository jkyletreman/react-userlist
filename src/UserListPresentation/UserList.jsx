import React from "react";
import UserData from "../context";

const UserList = ({ userData, selectUser}) => {
  return (
    <ul>
      {userData.map(user => {
        return (
          <li key={user.id} onClick={selectUser}>
            {user.first} {user.last}
          </li>
        );
      })}
    </ul>
  );
};

export default UserList;
