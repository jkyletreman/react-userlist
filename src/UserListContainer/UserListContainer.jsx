import React, { Component } from "react";
import UserList from "../UserListPresentation/UserList";
import UserInfoFull from "../UserInfoFullPresentation/UserInfoFull";
import UserData from "../context";
const userData = require("../data/userData.json");

export default class UserListContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userSelected: "",
      userIsSelected: false
    };
  }

  // filterUserData = username => {
  //   return userData.filter(user => `${user.first} ${user.last}` === username);
  // };

  render() {
    // const selectedUserData = this.state.userIsSelected
    //   ? this.filterUserData(this.state.userSelected)
    //   : null;
    //
    // const additionalUserInfo = this.state.userIsSelected ? (
    //   <UserInfoFullContainer selectedUser={selectedUserData[0]} />
    // ) : null;

    return (
      <UserData.Consumer>
        {context => (
          <React.Fragment>
            <UserList
              userData={context.state.userData}
              selectUser={context.selectUser}
            />
            <UserInfoFull
              userData={context.state.userData}
              selectUser={context.state.userSelected}
            />
          </React.Fragment>
        )}
      </UserData.Consumer>
    );
  }
}
