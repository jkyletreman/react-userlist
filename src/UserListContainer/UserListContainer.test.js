import React from "react";
import ReactDOM from "react-dom";
import UserListContainer from "./UserListContainer";
import UserList from "../UserListPresentation/UserList";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import ShallowRenderer from "react-test-renderer/shallow";

describe("<UserListContainer />", () => {
  it("renders user shallow correctly", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<UserListContainer />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  configure({ adapter: new Adapter() })

  it("should have a HOC ul with 3 li children", () => {
    const wrapper = shallow(<UserListContainer />)
    it('should have a ul with 3 children', () => {
      expect(wrapper.find('ul').dive().children().length).toBe(3);
    });
  });
});
