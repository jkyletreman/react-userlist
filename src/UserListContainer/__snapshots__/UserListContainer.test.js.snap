// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserListContainer /> renders user shallow correctly 1`] = `
<div>
  <UserList
    handleClick={[Function]}
    userData={
      Array [
        Object {
          "age": 51,
          "description": "John is a retired YouTuber",
          "first": "John",
          "id": 1,
          "last": "Roberts",
          "location": "Chicago, IL",
        },
        Object {
          "age": 27,
          "description": "Frita loves the desert, cheese, and turtles.",
          "first": "Frita",
          "id": 2,
          "last": "Waters",
          "location": "Phoenix, AZ",
        },
        Object {
          "age": 24,
          "description": "Bart skateboards everywhere.",
          "first": "Bart",
          "id": 3,
          "last": "Simpson",
          "location": "Springfield, IL",
        },
      ]
    }
  />
</div>
`;
